### PR TITLE
com.spotify:dns 3.2.2 which uses dnsjava 3.0.2

### PR DESCRIPTION
**Pull request by klaraward**
See internal PR for some context - https://ghe.spotify.net/java/bom/pull/538

### CODE REVIEW

File: folsom/pom.xml [hunk 1]
```diff
@@ -73,7 +73,7 @@
     </dependency>
     <dependency>
       <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
+      <artifacftId>junit</artifactId>
```
Comment (spkrka) [inline folsom/pom.xml]: This doesn't look correct

Comment (protocol7) [inline folsom/pom.xml]:
```suggestion
      <artifactId>junit</artifactId>
```

**Review by protocol7 — APPROVED** [review]

**Comment by protocol7** [comment]
@klaraward seems like com.spotify:dns:3.2.2 was built with Java 11, while Folsom requires 8. Could we build com.spotify:dns with 8 as well?

**Comment by klaraward** [comment]
@fdfzcq ?

**Comment by fdfzcq** [comment]
@klaraward @protocol7 I've changed maven plugin version from 9 to 8, will it do? https://github.com/spotify/dns-java/pull/47

**Comment by protocol7** [comment]
@fdfzcq yes, I think that would address this issue. Would you plan to release an update for com.spotify:dns with this fix?

**Comment by fdfzcq** [comment]
@protocol7 yes we plan to merge and release that PR today

**Comment by fdfzcq** [comment]
@protocol7 we merged the PR just to realize jdk 8 is not compatible with the latest `spotify/dns`. We will switch it back to jdk 9 for now and can look into moving it to jdk 8 (or multiple versions) later.

**Comment by protocol7** [comment]
@fdfzcq Thanks for the update.

**Comment by fdfzcq** [comment]
@klaraward @protocol7 we have now released newest version compiled with jdk 8: https://github.com/spotify/dns-java/releases/tag/v3.3.1

File: folsom/pom.xml [hunk 2]
```diff
@@ -50,7 +50,7 @@
     <dependency>
       <groupId>com.spotify</groupId>
       <artifactId>dns</artifactId>
-      <version>3.1.4</version>
+      <version>3.2.2</version>
```
Comment (protocol7) [inline folsom/pom.xml]:
```suggestion
      <version>3.3.1</version>
```

**Comment by protocol7** [comment]
@fdfzcq https://github.com/spotify/dns-java/pull/47 is a non-backwards compatible change in that it adds a new method to DnsSrvResolver. I don't know how common it would be to implement this among users of dns-java, but I could imagine it would be somewhat frequently implemented in tests. We can easily fix this in Folsom, but I would worry about rolling it, e.g. across the Spotify fleet.

**Comment by fdfzcq** [comment]
@protocol7 right, erik actually pointed out it which I thought I fixed but didn't https://github.com/spotify/dns-java/pull/47/files#r711963085 We would like to eventually move all users to use `resolveAsync` but for now this shouldn't break them.

**Comment by fdfzcq** [comment]
@protocol7 now third time charm :sweat_smile: https://github.com/spotify/dns-java/releases/tag/v3.3.2

File: folsom/pom.xml [hunk 3]
```diff
@@ -50,7 +50,7 @@
     <dependency>
       <groupId>com.spotify</groupId>
       <artifactId>dns</artifactId>
-      <version>3.1.4</version>
+      <version>3.3.1</version>
```
Comment (protocol7) [inline folsom/pom.xml]:
```suggestion
      <version>3.3.2</version>
```

**Comment by protocol7** [comment]
@fdfzcq that did indeed do the trick, thanks!

**Pull request merged**